### PR TITLE
[fix: #892] fix: enable to parse import modules using typescript-source-mode

### PR DESCRIPTION
**Pull request by euxn23**
Refered in #892 ,
typescript source mode can't use `import` statement to use other file,
throws `Error: TypeScript error: Cannot find module 'module-name'..`

I fixed this problem using `file full path` **instead of `baseName` and source file text** to TS Compiler API resolve module, not to readFile myself and delegate TS Compiler API.

I'm using typescript source mode in production, so I'd be grad if you merge as soon as possible.

### CODE REVIEW

**Comment by euxn23** [comment]
Dear @schani ,

Would you check this p-r please? I'm using this in production.

Kind regard.

**Comment by thegoldenmule** [comment]
I'm not terribly familiar with TS, but tests pass. Is this a breaking change @euxn23 ?

**Comment by euxn23** [comment]
No, this is not breaking change.

**Comment by mfrtrifork** [comment]
+1 I would like to have this feature.

**Comment by mihilmy** [comment]
This is a 1 year old PR. We really need this!

**Comment by mfrtrifork** [comment]
Dear @thegoldenmule and @untitaker 
What is the status of this PR? Is there someting preventing this from being merged?
Best regards, Michael

File: src/quicktype-typescript-input/index.ts [hunk 1]
```diff
@@ -18,93 +16,14 @@ const compilerOptions: ts.CompilerOptions = {
...
-    if (Array.isArray(sources)) {
-        fileNames = sources;
-        host = ts.createCompilerHost(compilerOptions);
-    } else {
-        fileNames = Object.getOwnPropertyNames(sources);
-        host = new CompilerHost(compilerOptions, sources);
-    }
-
-    const program = ts.createProgram(fileNames, compilerOptions, host);
+export function schemaForTypeScriptSources(sourceFileNames: string[]): JSONSchemaSourceData {
```
Comment (schani) [inline src/quicktype-typescript-input/index.ts:25]: Removing the `sources: { [fileName: string]: string }` argument overload breaks the quicktype web app.

**Review by schani — CHANGES_REQUESTED** [review]
This change breaks the quicktype API, in particular it will break the quicktype web app.

File: src/quicktype-typescript-input/index.ts (hunk 1 above)
Comment (dvdsgl) [reply src/quicktype-typescript-input/index.ts:25]: No longer used in web app

**Pull request merged**